esp/src/ganglia/GangliaWidget.js

```diff
@@ -0,0 +1,422 @@
+                        id: tabItem.name+idx,
+                        region: "center"
+                    });
+
+                    bordercontainer.addChild(contentpane);
+                    bordercontainer.addChild(toolbar);
+                    toolbar.addChild(button);
+                    bordercontainer.placeAt(context.tabContainer, idx);
+                    bordercontainer.startup();
+                });
+                context.tabContainer.selectChild(data.tabs[0].name.split(' ').join('_'));
+                /*TODO add interval refreshes of each tab
+                setInterval(function () {
+                    var node = dom.byId(tabItem.name+idx);
+                    query(".left", node).forEach(domConstruct.destroy);
+                    context._genGraphRefresh();
+                }, data.refreshInterval);*/
+            });
+        },
+
+        _genGraphRefresh : function (id) {
+            var context = this;
+            xhr("esp/files/ganglia/ganglia.json", {
+                handleAs: "json"
+            }).then(function (data){
+                arrayUtil.forEach(data.tabs, function(tabItem, idx) {
+                    arrayUtil.forEach(tabItem.metrics, function(metricItem) {
+                        arrayUtil.forEach(tabItem.time, function(timeItem) {
+                            var clean = metricItem.join('\n');
+                            context._calculateEpoch(timeItem)
+                            WsRrd.GangliaRRDGraphList({
+                                request:{
+                                    Clusters: "",
+                                    Servers: "",
+                                    RRDMetrics: clean,
+                                    StartTime: context.epochFilter,
+                                    EndTime: context.epochNow,
+                                    Width: data.width,
+                                    Height: data.height,
+                                    Title: timeItem
+                                },
+                            }).then(function (response) {
+                                if (lang.exists("GraphSVGDataResponse.Graph", response)) {
+                                    var graph = domConstruct.create("div", {
+                                        innerHTML: response.GraphSVGDataResponse.Graph,
+                                        class:'left'
+                                    }, tabItem.name+idx);
+                                }
+                            });
+                        });
+                    });
+                });
+            });
+        },
+
+        _onRefresh: function (id) {
+
+        },
+
+        _onFilterApply: function () {
+            var context = this;
+            var cluster = this.cluster;
+            var server = this.server;
+            var metrics = this.metrics;
+            var epochFilter = this.epochFilter;
+            var epochNow = this.epochNow;
```

Comment:
Good catch for repetitiveness. I also modified the alert message was trying to this.i18n instead of context.i18n and not displaying the string